core/src/main/java/nu/yona/server/analysis/service/WeekActivityDTO.java

```diff
@@ -84,4 +98,63 @@ static WeekActivityDTO createInstance(WeekActivity weekActivity, LevelOfDetail l
 								dayActivity -> DayActivityDTO.createInstance(dayActivity, levelOfDetail))),
 				weekActivity.hasPrevious(), weekActivity.hasNext());
 	}
+
+	public static WeekActivityDTO createInstanceInactivity(UserAnonymizedDTO userAnonymized, Goal goal, ZonedDateTime startOfWeek,
```

Comment:
I see you moved this logic from the entity to the DTO and pass the `missingInactivities` set to modify it. Is it in the end not cleaner to move this logic out of the DTO to the `ActivityService` class?